tools/analyzer_plugin/lib/src/diagnostic/missing_required_prop.dart

```diff
@@ -53,62 +54,122 @@ render() {
 // </editor-fold>
 
 class MissingRequiredPropDiagnostic extends ComponentUsageDiagnosticContributor {
-  @DocsMeta(_desc, details: _details, maturity: Maturity.experimental)
-  static const code = DiagnosticCode(
-    'over_react_required_prop',
-    "Missing required prop '{0}'.",
+  @DocsMeta(_desc, details: _details)
+  static const lateRequiredCode = DiagnosticCode(
+    'over_react_late_required_prop',
+    "Missing required late prop {0}.",
     AnalysisErrorSeverity.WARNING,
     AnalysisErrorType.STATIC_WARNING,
   );
 
+  // Note: this code is disabled by default in getDiagnosticContributors
+  @DocsMeta(_desc, details: _details)
+  static const annotationRequiredCode = DiagnosticCode(
+    'over_react_annotation_required_prop',
+    "Missing @requiredProp {0}.",
+    AnalysisErrorSeverity.INFO,
+    AnalysisErrorType.STATIC_WARNING,
+  );
+
+  static DiagnosticCode _codeForRequiredness(PropRequiredness requiredness) {
+    switch (requiredness) {
+      case PropRequiredness.late:
+        return lateRequiredCode;
+      case PropRequiredness.annotation:
+        return annotationRequiredCode;
+      case PropRequiredness.ignoredByConsumingClass:
+      case PropRequiredness.none:
+        throw ArgumentError('Only values considered required can be passed; got PropRequiredness.none', 'requiredness');
+    }
+  }
+
   @override
-  List<DiagnosticCode> get codes => [code];
+  List<DiagnosticCode> get codes => [
+        lateRequiredCode,
+        annotationRequiredCode,
+      ];
 
-  static final fixKind = FixKind(code.name, 200, "Add required prop '{0}'");
+  static final fixKind = FixKind('add_required_prop', 200, "Add required prop '{0}'");
 
   static final _debugCommentPattern = getDebugCommentPattern('over_react_required_props');
 
-  // final debugHelper = AnalyzerDebugHelper(result, collector, enabled: _debugCommentPattern.hasMatch(result.content));
+  /// Returns whether the debug helper should be enabled, caching results per result
+  /// so we don't have to run it in in every [computeErrorsForUsage].
+  ///
+  /// TODO Add shared debug flag computation in DiagnosticContributor/DiagnosticCollector
```

Comment:
Does this need to be addressed in this PR?

Reply:
No, but I would love to circle back to it later on, maybe next time we add a new debug flag